esx_service/vmdk_ops_test.py

```diff
@@ -532,10 +549,296 @@ def test_vmdkop_authorize(self):
+        self.assertEqual(None, error_info)
+
+        # create a volume with 500MB
+        opts={u'size': u'500mb', u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol3_name, opts)
+        self.assertEqual(None, error_info)
+
+        # add vm2 to tenant2
+        vm2_uuid = vmdk_utils.get_vm_uuid_by_name(self.vm2_name)
+        error_info = auth_api._tenant_vm_add(
+                                         name = self.tenant2_name, 
+                                         vm_list = [self.vm2_name]) 
+        self.assertEqual(None, error_info)
+
+        # set access privileges, create a volume with 100MB
+        rights = ["all"]
+        volume_maxsize = "500MB"
+        volume_totalsize = "2GB"
+        error_info = auth_api._tenant_access_add(name = self.tenant2_name, 
+                                                 datastore = self.datastore_name, 
+                                                 rights = rights, 
+                                                 volume_maxsize = volume_maxsize, 
+                                                 volume_totalsize = volume_totalsize)
+        # create a volume with default size
+        opts={u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm2_uuid, self.vm2_name, self.vm2_config_path, auth.CMD_CREATE, self.tenant2_vol1_name, opts)
+        self.assertEqual(None, error_info)
+        
+        # create a volume with 500MB
+        opts={u'size': u'500MB', u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm2_uuid, self.vm2_name, self.vm2_config_path, auth.CMD_CREATE, self.tenant2_vol2_name, opts)
+        self.assertEqual(None, error_info)
+
+        # create a volume with 500MB
+        opts={u'size': u'500mb', u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm2_uuid, self.vm2_name, self.vm2_config_path, auth.CMD_CREATE, self.tenant2_vol3_name, opts)
+        self.assertEqual(None, error_info)
+        
+        # list volumes for tenant1
+        opts = {}
+        result = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_LIST, None, opts)
+       
+        self.assertEqual(3, len(result))
+        self.assertEqual("tenant1_vol1", result[0]['Name'])
+        self.assertEqual("tenant1_vol2", result[1]['Name'])
+        self.assertEqual("tenant1_vol3", result[2]['Name'])
+
+        # list volumes for tenant2
+        opts = {}
+        result = vmdk_ops.executeRequest(vm2_uuid, self.vm2_name, self.vm2_config_path, auth.CMD_LIST, None, opts)
+       
+        self.assertEqual(3, len(result))
+        self.assertEqual("tenant2_vol1", result[0]['Name'])
+        self.assertEqual("tenant2_vol2", result[1]['Name'])
+        self.assertEqual("tenant2_vol3", result[2]['Name'])
+
+    
```

Comment:
A general comment on tests: it is better to have them data drivel . I.e. instead of using tenantx_volx ad opts and other stuff in the code, ALL that should be in data structures and the test should just taverse the structures. See example (testInfo traversing) in vmdk_ops_test.py.

It's OK to leave it as is for this PR but please do open an issue for test rearrangement and making them data driven - that allows to understand tests better and to add to the test set easily